src/main/java/frc/robot/poseestimator/GBPoseEstimator.java

```diff
@@ -0,0 +1,191 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.interpolation.TimeInterpolatableBuffer;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.math.kinematics.SwerveDriveWheelPositions;
+import frc.robot.subsystems.GBSubsystem;
+import frc.robot.vision.limelights.GyroAngleValues;
+import frc.robot.vision.limelights.LimelightFilterer;
+import frc.robot.poseestimator.observations.OdometryObservation;
+import frc.robot.poseestimator.observations.VisionObservation;
+import frc.robot.vision.limelights.LimelightFiltererConfig;
+import frc.utils.Conversions;
+import org.littletonrobotics.junction.Logger;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class GBPoseEstimator extends GBSubsystem implements IPoseEstimator {
+
+	private final TimeInterpolatableBuffer<Pose2d> odometryPoseInterpolator;
+	private final TimeInterpolatableBuffer<Pose2d> estimatedPoseInterpolator;
+	private final SwerveDriveKinematics kinematics;
+	private final double[] odometryStandardDeviations;
+	private Pose2d odometryPose;
+	private Pose2d estimatedPose;
+	private SwerveDriveWheelPositions latestWheelPositions;
+	private Rotation2d latestGyroAngle;
+	private final LimelightFilterer limelightFilterer;
+
+	public GBPoseEstimator(
+		String logPath,
+		SwerveDriveKinematics kinematics,
+		SwerveDriveWheelPositions initialWheelPositions,
+		Rotation2d initialGyroAngle,
+		double[] odometryStandardDeviations,
+		LimelightFiltererConfig limelightFiltererConfig,
+		Pose2d initialRobotPose
+	) {
+		super(logPath);
+		this.odometryPose = new Pose2d();
+		this.estimatedPose = new Pose2d();
+		this.odometryPoseInterpolator = TimeInterpolatableBuffer.createBuffer(PoseEstimatorConstants.POSE_BUFFER_SIZE_SECONDS);
+		this.estimatedPoseInterpolator = TimeInterpolatableBuffer.createBuffer(PoseEstimatorConstants.POSE_BUFFER_SIZE_SECONDS);
+		this.kinematics = kinematics;
+		this.latestWheelPositions = initialWheelPositions;
+		this.latestGyroAngle = initialGyroAngle;
+		this.odometryStandardDeviations = new double[PoseArrayEntryValue.POSE_ARRAY_LENGTH];
+		this.limelightFilterer = new LimelightFilterer(limelightFiltererConfig, this::getEstimatedPoseAtTimeStamp);
```

Comment:
get this as param! (i think raz gave a comment here too)

Reply:
get what as a param?

I cant get the Limelight filterer as a param

Reply:
why?